interface-definitions/include/firewall/firewall-hashing-parameters.xml.i

```diff
@@ -0,0 +1,35 @@
+<!-- include start from firewall/firewall-hashing-parameters.xml.i -->
+<leafNode name="hash">
+  <properties>
+    <help>Define the parameters of the packet header to apply the hashing, or choose random</help>
+    <completionHelp>
+      <list>source-address destination-address source-port destination-port random</list>
+    </completionHelp>
+    <valueHelp>
+      <format>source-address</format>
+      <description>Use source IP address for hashing</description>
+    </valueHelp>
+    <valueHelp>
+      <format>destination-address</format>
+      <description>Use destination IP address for hashing</description>
+    </valueHelp>
+    <valueHelp>
+      <format>source-port</format>
+      <description>Use source port for hashing</description>
+    </valueHelp>
+    <valueHelp>
+      <format>destination-port</format>
+      <description>Use destination port for hashing</description>
+    </valueHelp>
+    <valueHelp>
+      <format>random</format>
+      <description>Do not use information from ip header. Use random value.</description>
+    </valueHelp>
+    <constraint>
+      <regex>(source-address|destination-address|source-port|destination-port|random)</regex>
+    </constraint>
+    <multi/>
```

Comment:
Does it make sense to have a hash value based on multiple configurable sources?

Reply:
From my point of view, it does make sense.
- More flexibility
- If more parameters are used, it easier to get different values of hashing for every different connection. In consequence, better load balance distribution.

For example if hash is only calculated using source IP address (think on a LAN, applying source nat to reach internet through 2 uplinks), then all connection form host A will always use the same uplink... If more parameters are included in the hashing, then connections from same host will be balanced.